lib/order.rb

```diff
@@ -0,0 +1,109 @@
+class Order
+  
+  attr_reader :id, :products, :customer, :fulfillment_status
+  # @products have string keys, not symbol
+  
+  def initialize(id, products, customer, fulfillment_status=:pending)
+    # assuming arguments id/products/customers are all legit
+    @id = id
+    @products = products
+    @customer = customer
+    
+    acceptable_fulfillment_stats = [:pending, :paid, :processing, :shipped, :complete]
+    if acceptable_fulfillment_stats.include? fulfillment_status
+      @fulfillment_status = fulfillment_status
+    else
+      raise ArgumentError, "Invalid fulfillment status!"
```

Comment:
It's helpful to include what status was invalid in the message.

```suggestion
      raise ArgumentError, "#{fulfillment_status} is an invalid fulfillment status!"
```